lesson17/index.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+
+const resourceUrl = "https://jsondata.okiba.me/v1/json/9gjw3210424132243";
+const body = document.getElementById("body");
+const wrapper = document.querySelector(".wrapper");
+const box = document.querySelector(".box");
+const prevButton = document.querySelector(".prevNext__prev");
+const nextButton = document.querySelector(".prevNext__next");
+const navigationNum = document.querySelector(".navigation__num");
+const fragment = document.createDocumentFragment();
+let li;
+let imageLength;
+let currentNum = 0;
+
+const timeout = (ms) => {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
+
+const request = async () => {
+  const resource = await fetch(resourceUrl);
+  let json = await resource.json();
+  return json;
+};
+
+const fetchData = async () => {
+  let res;
+  try {
+    await timeout(3000);
+    res = await request();
+  } catch (error) {
+    res = [];
+  } finally {
+    return res;
+  }
+};
+
+const init = async () => {
+  const datas = await fetchData();
+  if (datas.length === 0) {
+    box.textContent = "コンテンツがありません";
+    return;
+  }
+  createImagesView(datas);
+};
+
+const createImages = (image, index) => {
+  li = document.createElement("li");
+  const img = document.createElement("img");
+  img.classList.add("image");
+  img.setAttribute("src", image.src);
+  img.setAttribute("alt", image.alt);
+  li.classList.add("listItem");
+  li.dataset.num = index; // HACK: data属性にindexをそのまま用いているがこれは適切か。
```

Comment:
使ってもいいですが、
```suggestion
  li.dataset.index= index; // HACK: data属性にindexをそのまま用いているがこれは適切か。
```
numではないんじゃないですかね。なんのnumだかわからないです

Reply:
修正しました。 7715546